neon/data/add_testing_repo.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+#
+# Copyright (C) 2018 Bhushan Shah <bshah@kde.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License or (at your option) version 3 or any later version
+# accepted by the membership of KDE e.V. (or its successor approved
+# by the membership of KDE e.V.), which shall act as a proxy
+# defined in Section 14 of version 3 of the license.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# add /testing repo and apt update/upgrades
+
+puts "#{$0}: Adding the testing repository to verify the upgrade"
+
+ENV['DEBIAN_FRONTEND'] = 'noninteractive'
+
+File.write('/etc/apt/preferences.d/neon-testing', <<-PREFERENCE)
+Package: *
+Pin: release l=Neon - Testing
+Pin-Priority: 1001
+PREFERENCE
+
+system('add-apt-repository "deb https://archive.neon.kde.org/testing xenial main"') || raise
```

Comment:
not sure the hardcoding of xenial is particularly lovely given xenial will be dead in 2 months or so